### PR TITLE
Allow using different mock namespaces

### LINKED ISSUE #12605: Ensure that the Eventing fast-integration tests can run on a cluster with pre-existing test setup
**Description**

We have automated most of our manual release tests, and have added them to the fast-integration test suites. However, running them on a release candidate automatically when a release candidate cluster is produced, [have been disabled](https://github.com/kyma-project/test-infra/pull/4273). Therefore, we need to manually run these tests on a release candidate cluster. To do this, we need to verify, whether running these tests on a cluster with existing manual tests and setups, would cause interference or could always run.

### PR DESCRIPTION
**Pull request by pxsalehi**
**Description**

The existing Commerce Mock fixture uses a fixed host value for the APIRule. 

Changes proposed in this pull request:

- Parametrize the Commerce Mock objects used in the test, to allow co-existing mocks in different namespaces.
- Make sure the mock namespace is passed around every where in the test. Currently in many places the value "mocks" is hardcoded!

**Related issue(s)**

Closes #12605

### CODE REVIEW

**Review by mfaizanse — DISMISSED** [review]

File: tests/fast-integration/README.md [hunk 1]
```diff
@@ -63,6 +63,13 @@ npm install
 npm run test-eventing
 ```
 
+**NOTE:** You can change the namespace used for the Commerce Mock objects using:
+```
+MOCK_NAMESPACE=mock2 make ci-test-eventing
+```
+Using a different namespace changes the host used for the Commerce Mock APIRule as well. 
+This allows different Commerce Mocks to co-exist in different namespaces.
```
Comment (majakurcius) [inline tests/fast-integration/README.md:71]:
```suggestion
> **NOTE:** You can change the Namespace used for the Commerce Mock objects using:
> ```
> MOCK_NAMESPACE=mock2 make ci-test-eventing
> ```
> Using a different Namespace changes the host used for the Commerce Mock APIRule as well. 
> This allows different Commerce Mocks to co-exist in different Namespaces.
```

As per our guidelines, we [capitalize `Namespace`](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#capitalized-terms) and [put the NOTE, CAUTION, and TIP notes in panels](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#panels). Please check the guidelines for reference.

**Review by mfaizanse — APPROVED** [review]

**Review by majakurcius — APPROVED** [review]

**Review by wozniakjan — APPROVED** [review]
/lgtm

**Pull request merged**